platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/utility/RecycleBinImpl.java

```diff
@@ -16,48 +16,108 @@
 
 package com.swirlds.common.io.utility;
 
+import static com.swirlds.common.io.utility.FileUtils.deleteDirectory;
 import static com.swirlds.logging.LogMarker.EXCEPTION;
+import static com.swirlds.logging.LogMarker.STARTUP;
 
+import com.swirlds.base.state.Startable;
+import com.swirlds.base.state.Stoppable;
+import com.swirlds.base.time.Time;
 import com.swirlds.common.config.StateConfig;
 import com.swirlds.common.io.config.RecycleBinConfig;
+import com.swirlds.common.metrics.IntegerGauge;
+import com.swirlds.common.metrics.Metrics;
 import com.swirlds.common.system.NodeId;
+import com.swirlds.common.threading.framework.StoppableThread;
+import com.swirlds.common.threading.framework.config.StoppableThreadConfiguration;
 import com.swirlds.common.threading.locks.AutoClosableLock;
 import com.swirlds.common.threading.locks.Locks;
 import com.swirlds.common.threading.locks.locked.Locked;
+import com.swirlds.common.threading.manager.ThreadManager;
+import com.swirlds.common.utility.CompareTo;
 import com.swirlds.config.api.Configuration;
 import edu.umd.cs.findbugs.annotations.NonNull;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.time.Duration;
+import java.time.Instant;
 import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 /**
  * A standard implementation of a {@link RecycleBin}.
  */
-class RecycleBinImpl implements RecycleBin {
+public class RecycleBinImpl implements RecycleBin, Startable, Stoppable {
 
     private static final Logger logger = LogManager.getLogger(RecycleBinImpl.class);
 
+    private final Time time;
     private final Path recycleBinPath;
+    private final Duration maximumFileAge;
+    private int recycledFileCount;
```

Comment:
Probably worth noting that this isn't really a file count, rather a count of files and directories directly in the recycle bin directory

Reply:
Changed name to `topLevelRecycledFileCount` and added documentation.